tests/unit/api/bucketDeleteEncryption.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+
+const { bucketPut } = require('../../../lib/api/bucketPut');
+const bucketPutEncryption = require('../../../lib/api/bucketPutEncryption');
+const bucketDeleteEncryption = require('../../../lib/api/bucketDeleteEncryption');
+const { cleanup, DummyRequestLogger, makeAuthInfo } = require('../helpers');
+const { templateSSEConfig, templateRequest, getSSEConfig } = require('../utils/bucketEncryption');
+const log = new DummyRequestLogger();
+const authInfo = makeAuthInfo('accessKey1');
+const bucketName = 'bucketname';
+
+const bucketPutRequest = {
+    bucketName,
+    headers: { host: `${bucketName}.s3.amazonaws.com` },
+    url: '/',
+};
+
+describe('bucketDeleteEncryption API', () => {
+    before(() => cleanup());
+
+    beforeEach(done => bucketPut(authInfo, bucketPutRequest, log, done));
+    afterEach(() => cleanup());
+
+    it('should perform a no-op if no sse config exists', done => {
+        bucketDeleteEncryption(authInfo, templateRequest(bucketName, {}), log, err => {
+            assert.ifError(err);
+            return getSSEConfig(bucketName, (err, sseInfo) => {
+                assert.ifError(err);
+                assert.strictEqual(sseInfo, null);
+                done();
+            });
+        });
+    });
+
+    it('should disable mandatory sse for AES256', done => {
+        const post = templateSSEConfig({ algorithm: 'AES256' });
+        bucketPutEncryption(authInfo, templateRequest({ post }), log, err => {
+            assert.ifError(err);
+            bucketDeleteEncryption(authInfo, templateRequest({}), log, err => {
+                assert.ifError(err);
+                return getSSEConfig(bucketName, (err, sseInfo) => {
+                    assert.ifError(err);
+                    assert.strictEqual(sseInfo.mandatory, false);
+                    done();
+                });
+            });
+        });
+    });
+
+    it('should disable mandatory sse for aws:kms', done => {
```

Comment:
You might want to merge this test with the preceeding one and a `forEach` on the algorithm since the logic is essentially the same